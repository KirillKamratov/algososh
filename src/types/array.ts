import { ElementStates } from "./element-states";

export type TArray = {
  value: string;
  state: ElementStates
}
