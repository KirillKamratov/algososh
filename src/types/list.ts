import { ElementStates } from "./element-states";


export interface IList<T> {
  append: (element: T) => void;
  prepend: (element: T) => void;
  getElementByIndex: (index: number) => T | null;
  addByIndex: (element: T, index: number) => void;
  removeByIndex: (index: number) => void;
  removeHead: () => void;
  removeTail: () => void;
  getSize: () => number;
  toArray: () => T[];
}

export interface IInitialList {
  value: string;
  state: ElementStates;
  head?: string;
  tail?: string;
}



