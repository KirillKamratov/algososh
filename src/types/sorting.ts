import { ElementStates } from "./element-states";

export type TSorting = {
  value: number;
  state: ElementStates;
}
