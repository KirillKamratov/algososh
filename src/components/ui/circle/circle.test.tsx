import React from 'react';
import {Circle} from "./circle";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import {ElementStates} from "../../../types/element-states";

describe('Testing circle component', () => {
  it('circle with letters', () => {
    const tree = renderer.create( <Circle letter={'ABC'} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('circle without letters', () => {
    const tree = renderer.create(<Circle />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('circle with head', () => {
    const tree = renderer.create( <Circle head={'head'} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('circle with react-element head', () => {
    const tree = renderer.create( <Circle head={<Circle isSmall={true} />} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('circle with tail', () => {
    const tree = renderer.create( <Circle tail={'tail'} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('circle with react-element tail', () => {
    const tree = renderer.create( <Circle tail={<Circle isSmall={true} />} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('circle with index', () => {
    const tree = renderer.create( <Circle index={0} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('small circle', () => {
    const tree = renderer.create( <Circle isSmall={true} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('circle in default state', () => {
    const tree = renderer.create( <Circle state={ElementStates.Default}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('circle in changing state', () => {
    const tree = renderer.create( <Circle state={ElementStates.Changing} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('circle in modified state', () => {
    const tree = renderer.create( <Circle state={ElementStates.Modified} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
