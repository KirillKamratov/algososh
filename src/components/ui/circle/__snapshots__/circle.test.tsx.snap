// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing circle component circle in changing state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle component circle in default state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle component circle in modified state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle component circle with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle component circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Testing circle component circle with letters 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle component circle with react-element head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle component circle with react-element tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Testing circle component circle with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Testing circle component circle without letters 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle component small circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
